### ISSUE TITLE
[BUG] Not seeing custom metrics in Application Insights when using OpenTelemetry Azure Monitor exporter

### ISSUE THREAD

**Issue by agr** [Azure/azure-sdk-for-net#38352]
### Library name and version

Azure.Monitor.OpenTelemetry.Exporter 1.0.0-beta.14

### Describe the bug

I have an Azure function set up the following way (using [`OpenTelemetry.Extensions.Hosting 1.4.0`](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting/1.4.0)):

Startup class:
```C#
        protected void ConfigureJobServices(IServiceCollection services, IConfiguration configuration)
        {
            var aiConnectionString = configuration["APPLICATIONINSIGHTS_CONNECTION_STRING"];
            var p1 = "...";
            var p2 = "....";
            services.AddOpenTelemetry()
                .ConfigureResource(builder => builder.AddService(serviceName: "MyService"))
                .WithMetrics(builder => SetupMeterProviderBuilder(builder, aiConnectionString, p1, p2));
        }

        private static void SetupMeterProviderBuilder(
            MeterProviderBuilder meterBuilder,
            string aiConnectionString,
            string p1, 
            string p2)
        {
            meterBuilder
                .AddMeter(TelemetryService.MeterName);

            if (!string.IsNullOrWhiteSpace(aiConnectionString))
            {
                meterBuilder
                    .AddAzureMonitorMetricExporter(options => options.ConnectionString = aiConnectionString);
            }

            if (!string.IsNullOrWhiteSpace(p1)
                && !string.IsNullOrWhiteSpace(p2))
            {
                meterBuilder
                    .AddXXXXMetricExporter(options => options.ConnectionString = $"...{p1};{p2}...");
            }
        }
```

That `AddXXXXMetricExporter` is an internal service to which we also want to dump metrics.

Next, we have actual metric reporting:

```C#
    public class TelemetryService
    {
        public const string MeterName = "MyServiceMeter";

        private readonly Meter _meter;
        private readonly Histogram<double> _metric1;

        public TelemetryService()
        {
            _meter = new Meter(MeterName);
            _metric1 = _meter.CreateHistogram<double>("MetricName", unit: "day");
        }

        public void TrackMetric1(TimeSpan duration, string p1, string p2)
        {
            _timeToExpiration.Record(duration.TotalDays, new[]
            {
                KeyValuePair.Create<string, object>("Dimension1", $"{keyVaultName}/{secretName}")
            });
        }
    }
```

### Expected behavior

I expect metrics to appear in customMetrics table in AI as well as our internal metric service.

### Actual behavior

I only see metrics in our internal system. I only see standard Function metric there which, I believe, are not exported through OpenTelemetry but by interacting with AI client libraries directly.

I tried different kinds of metrics besides Histogram (Counter, UpDownCounter), the behavior is the same in all cases: I see them in the internal system, but not in AI.

### Reproduction Steps

Create an Azure Application Insights instance (or use existing one), grab its connection string.
Create new Azure Function project in VS with the following settings:

![Project settings](https://github.com/Azure/azure-sdk-for-net/assets/102933/a43c2e5d-230f-4950-9c09-3aba12c735cf)

Install extra packages:

```
dotnet add package OpenTelemetry.Extensions.Hosting -v 1.4.0
dotnet add package Azure.Monitor.OpenTelemetry.Exporter -v 1.0.0-beta.14
```

Create a `Startup` class (update the value of `aiConnectionString ` variable to the AI connection string grabbed earlier):

```C#
[assembly: FunctionsStartup(typeof(Startup))]

namespace MyNamespace
{
    public class Startup : FunctionsStartup
    {
        public override void Configure(IFunctionsHostBuilder builder)
        {
            // AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);
            ConfigureJobServices(builder.Services, builder.GetContext().Configuration);
        }

        protected void ConfigureJobServices(IServiceCollection services, IConfiguration configuration)
        {
            var aiConnectionString = "InstrumentationKey=<AI instrumentation key goes here>";
            services.AddOpenTelemetry()
                .ConfigureResource(builder => builder.AddService(serviceName: "MyService"))
                .WithMetrics(builder => 
                    builder
                        .AddMeter("MyMeter")
                        .AddAzureMonitorMetricExporter(options => options.ConnectionString = aiConnectionString));
        }
    }
}
```

Update class containing the function trigger:

```C#
    public class MyFunction
    {
        private readonly ILogger<MyFunction> _logger;

        public MyFunction(
            ILogger<MyFunction> logger)
        {
            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
        }

        [FunctionName("MyFunction")]
        public async Task Run([TimerTrigger("0 */5 * * * *", RunOnStartup = true)] TimerInfo myTimer)
        {
            _logger.LogInformation("Running function");
            var m = new Meter("MyMeter");
            var metric = m.CreateHistogram<double>("Metric1");
            metric.Record(123.456);
        }
    }
```

Deploy to Azure, check application insights for the presence of "Metric1".

### Environment

Azure Functions v4 targeting .net6 non-isolated.

**Comment by agr** [comment]
Hm... Apparently, setting the service name in `.ConfigureResource(builder => builder.AddService(serviceName: "MyService"))` changes the `cloud_RoleName` to `MyService` and I was filtering AI wrong. So at least, the code from the "Reproduction Steps" works as expected, which means in the original code the problem is likely with me doing something wrong.

**Comment by agr** [comment]
...and in the original code I only added `.ConfigureResource` after I started trying to figure out where did my metrics go. It seems I never actually deployed it and when it is not specified `cloud_RoleName` was defaulting to `unknown_service:w3wp`, this is where I found all the metrics reported in the past.

So, apologies, there are no issues with metric reporting, just unexpected value of `cloud_RoleName` is used.
